plugins/orchestrator/src/components/next/ExecuteWorkflowPage.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAsync } from 'react-use';
+
+import { InfoCard, Progress } from '@backstage/core-components';
+import {
+  useApi,
+  useRouteRef,
+  useRouteRefParams,
+} from '@backstage/core-plugin-api';
+import { JsonValue } from '@backstage/types';
+
+import { Button, Grid, Typography } from '@material-ui/core';
+
+import { WorkflowDataInputSchemaResponse } from '@janus-idp/backstage-plugin-orchestrator-common';
+
+import { orchestratorApiRef } from '../../api';
+import {
+  executeWorkflowRouteRef,
+  workflowInstanceRouteRef,
+} from '../../routes';
+import { BaseOrchestratorPage } from './BaseOrchestratorPage';
+
+export interface ExecuteWorkflowPageProps {
+  initialState?: Record<string, JsonValue>;
+}
+
+export const ExecuteWorkflowPage = (props: ExecuteWorkflowPageProps) => {
+  const orchestratorApi = useApi(orchestratorApiRef);
+  const { workflowId } = useRouteRefParams(executeWorkflowRouteRef);
+  const [isExecuting, setIsExecuting] = useState(false);
+
+  // @ts-ignore:
+  const [formState, setFormState] = useState(props.initialState);
+
+  const navigate = useNavigate();
+  const instanceLink = useRouteRef(workflowInstanceRouteRef);
+
+  const { value, error, loading } =
+    useAsync(async (): Promise<WorkflowDataInputSchemaResponse> => {
+      const response =
+        await orchestratorApi.getWorkflowDataInputSchema(workflowId);
+      return response;
+    }, [orchestratorApi, workflowId]);
+
+  const isReady = useMemo(() => !loading && !error, [loading, error]);
+
+  const onExecute = useCallback(async () => {
+    const parameters: Record<string, JsonValue> = {};
+    if (value?.schema && formState) {
+      for (const key in formState) {
+        if (formState.hasOwnProperty(key)) {
+          const property = formState[key];
+          Object.assign(parameters, property);
+        }
+      }
+    }
+
+    setIsExecuting(true);
+    try {
+      const response = await orchestratorApi.executeWorkflow({
+        workflowId,
+        parameters,
+      });
+      navigate(instanceLink({ instanceId: response.id }));
+    } catch (err) {
+      // eslint-disable-next-line
```

Comment:
Yes, I agree, we need a unified error handling, we can discuss with UXD about this, but I think this should be handled by another task, since it is related to all error handling. WDYT?

Reply:
Sure, +1. Please, add a task to FLPATH-558 about this topic